cmd/ci_trace.go

```diff
@@ -185,6 +190,7 @@ func filterJobArg(args []string) (string, []string, error) {
 
 func init() {
 	ciTraceCmd.Flags().Bool("merge-request", false, "use merge request pipeline if enabled")
```

Comment:
That's something I would need to dig further and see how to implement. The idea of having `--merge-request` was to avoid ambiguity between branch name and MR numbers in the CLI cmdline, because it's not difficult to find projects where developers use the ticket/issue number as their branch names. So instead of making lab "too smart" we preferred to let the user take over the action completely. But I'll give it a try and see if we can get anything better done. This MR vs branch vs ... code is indeed messy.

The suggestion about the `if enabled` is indeed correct I'll make sure to re-visit it.